.github/workflows/build-test-docs.yml

```
@@ -0,0 +1,78 @@
+name: Build Test and Documentation
+# Build on all branches
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
```

```suggestion
on:
  pull_request:
  push:
    branches:
      - master
```
This will then build on merge to master and on pr but won't try to do both when someone submits a pr

---

Yes, good suggestion! It is included in the 3 separate workflow files now.